src/mlpack/core/data/split_data.hpp

```diff
@@ -368,6 +341,241 @@ Split(const arma::Mat<T>& input,
                          std::move(testData));
 }
 
+/**
+ * Given an input dataset and labels, split into a training set and test set.
+ * Example usage below.  This overload places the split dataset into the four
+ * output parameters given (trainData, testData, trainLabel, and testLabel).
+ *
+ * The input dataset must be of type arma::field. It should have the shape -
+ * (n_rows = 1, n_cols = Number of samples, n_slices = 1)
+ *
+ * NOTE: Here FieldType could be arma::field<arma::mat> or arma::field<arma::vec>
+ *
+ * @code
+ * arma::field<arma::mat> input = loadData();
+ * arma::field<arma::vec> label = loadLabel();
+ * arma::field<arma::mat> trainData;
+ * arma::field<arma::mat> testData;
+ * arma::field<arma::vec> trainLabel;
+ * arma::field<arma::vec> testLabel;
+ * math::RandomSeed(100); // Set the seed if you like.
+ *
+ * // Split the dataset into a training and test set, with 30% of the data being
+ * // held out for the test set.
+ * Split(input, label, trainData,
+ *                testData, trainLabel, testLabel, 0.3);
+ * @endcode
+ *
+ * @param input Input dataset to split.
+ * @param inputLabel Input labels to split.
+ * @param trainData FieldType to store training data into.
+ * @param testData FieldType test data into.
+ * @param trainLabel Field vector to store training labels into.
+ * @param testLabel Field vector to store test labels into.
+ * @param testRatio Percentage of dataset to use for test set (between 0 and 1).
+ * @param shuffleData If true, the sample order is shuffled; otherwise, each
+ *       sample is visited in linear order. (Default true.)
+ */
+template <typename FieldType,
+          typename = std::enable_if_t<
+              arma::is_Col<typename FieldType::object_type>::value ||
+              arma::is_Mat_only<typename FieldType::object_type>::value>>
+void Split(FieldType& input,
+           arma::field<arma::vec>& inputLabel,
+           FieldType& trainData,
+           arma::field<arma::vec>& trainLabels,
```

Comment:
Can we be generic about this too?  Should the user be able to pass, e.g., `arma::field<>` with an `arma::Row<size_t>` or anything?

Reply:
I remember, earlier I fixed it to `arma::vec` keeping in mind the bounding box problem that we were trying to solve in the original issue. But yes, you are right. This could be easily generalized to `arma::field<T>` by introducing another template parameter. I will give it a try and see if it works well. (Recently I have been struggling with many errors related to C++ templates and have started to get nighmares from them :p). Lets see how this one goes :-)

Reply:
I know the feeling; template error messages are not exactly friendly. :smile: (My strategy is just to start with the first one; basically everything past that tends to be useless noise...)